src/main.py

```diff
@@ -1,8 +1,40 @@
 """The main pipeline"""
 
-from src.utils.testfunctions import add
+import toml
+from utils.testfunctions import add
 
+def toml_parser():
+    """Function to parse the userconfig.toml file.
+
+    Returns:
+        A dictionary where the keys are section titles within the TOML file.
+        If only one variable under the section title in the TOML file is given
+        then it is passed directly as a dictionary value. If more than one
+        variable is defined then they are parsed as a dictionary themselves.
+        An example of what is returned is given below:
+        
+        {'title': 'TOML Example config', 'period': {'start_period': 
+        datetime.date(1990, 10, 10), 'end_period': datetime.date(2000, 10, 5)}, 
+        'source_file': {'location': 'D:/Data', 'fileName': 'file.txt'}, 'output_location': 
+        {'hive_db': 'hive.db', 'tableName': 'name.table'}, 'outlier_correction': 
+        {'location': 'D:/', 'fileName': 'outliers.txt', 'bool': True}}
+    """    
+    return toml.load("/home/cdsw/research-and-development/config/userconfig.toml")
+
+def period_select():
+    """Function returning the start and end date under consideration.
+
+    Returns:
+        A tuple containing two datetime.date objects. The first is the
+        start date of the period under consideration, the second is the
+        end date of that period. 
+        Example:
+    
+        (datetime.date(1990, 10, 10), datetime.date(2000, 10, 5))
+    """    
+    period_dict = toml_parser()["period"]
+    return period_dict["start_period"], period_dict["end_period"]
```

Comment:
It's not a strict rule, but I suggest leaving a blank line between your process code and the return statement.

Reply:
Latest commit creates a blank line between each return statement and its corresponding process code.